client/issues-lib.js

```diff
@@ -8,133 +9,148 @@ if (window.Issue && window.IssueList) {
 // https://www.google.com/design/spec/style/color.html#color-color-palette
 var _reviewLevelMetadata = {
   'daily': {
-    query: {label: 'Update-daily'},
+    query: {label: 'Update-Daily'},
     outOfSLOColor: '#B71C1C',  // Red 900
   },
   'weekly': {
-    query: {label: 'Update-weekly'},
+    query: {label: 'Update-Weekly'},
     outOfSLOColor: '#C62828',  // Red 800
   },
   'fortnightly': {
-    query: {label: 'Update-fortnightly'},
+    query: {label: 'Update-Fortnightly'},
     outOfSLOColor: '#D32F2F',  // Red 700
   },
   'monthly': {
-    query: {label: 'Update-monthly'},
+    query: {label: 'Update-Monthly'},
     outOfSLOColor: '#E53935',  // Red 600
   },
   'quarterly': {
-    query: {label: 'Update-quarterly'},
+    query: {label: 'Update-Quarterly'},
     outOfSLOColor: '#F44336',  // Red 500
   },
   'none (P0)': {
-    query: {label: 'Pri-0', '-has': 'update'},
+    query: {label: 'Pri-0', '-has': 'Update'},
   },
   'none (P1)': {
-    query: {label: 'Pri-1', '-has': 'update'},
+    query: {label: 'Pri-1', '-has': 'Update'},
   },
   'none (P2)': {
-    query: {label: 'Pri-2', '-has': 'update'},
+    query: {label: 'Pri-2', '-has': 'Update'},
   },
   'none (P3)': {
-    query: {label: 'Pri-3', '-has': 'update'},
+    query: {label: 'Pri-3', '-has': 'Update'},
   },
 };
 var _defaultReviewLevel = 'none';
 var _inSLOColor = '#4CAF50';  // Green 500
 var _noSLOColor = '#757575';  // Grey 600
 
-var Issue = function(monorailIssue) {
-  this._rawData = monorailIssue;
-
-  this.id = monorailIssue.id;
-  if (monorailIssue.owner) {
-    this.owner = monorailIssue.owner.name;
-  } else {
-    this.owner = null;
-  }
-  this.summary = monorailIssue.summary;
-  this.priority = undefined;
-  this._reviewLevel = _defaultReviewLevel;
-
-  this._lastUpdatedString = monorailIssue.updated;
-  this._lastUpdatedMS = Date.parse(this._lastUpdatedString);
-
-  for (var i = 0; i < monorailIssue.labels.length; i++) {
-    var label = monorailIssue.labels[i];
-    if (label.substring(0, 4) == 'Pri-') {
-      this.priority = Number(label.substring(4));
+class Issue {
+  constructor(params) {
+    // Copy constructor
+    if (params instanceof Issue) {
```

Comment:
If this is only used by clone, then it's simpler to directly construct around a copy of the params in clone than to switch every time an issue is constructed.